contracts/serverless-v2/Relay.sol

```diff
@@ -1052,7 +1087,22 @@ contract Relay is
         // TODO: do we need to check this bool success
         (bool success, ) = _jobOwner.call{value: callbackAmount}("");
 
-        emit JobSubscriptionTerminated(_jobSubsId, _jobOwner, usdcAmount, callbackAmount, success);
+        emit JobSubscriptionFundsWithdrawn(_jobSubsId, _jobOwner, usdcAmount, callbackAmount, success);
+    }
+
+    function _terminateJobSubscription(
+        uint256 _jobSubsId,
+        address _jobOwner
+    ) internal {
+        if(jobSubscriptions[_jobSubsId].job.jobOwner != _jobOwner)
+            revert RelayNotJobSubscriptionOwner();
+
+        if(block.timestamp > jobSubscriptions[_jobSubsId].terminationTimestamp)
+            revert RelayJobSubscriptionTerminated();
+
+        jobSubscriptions[_jobSubsId].terminationTimestamp = block.timestamp;
```

Comment:
it should be block.timestamp + overalltimeout, this is what we set as lower limit in updateTerminationTimestamp.